activerecord/lib/active_record/associations/association.rb

```diff
@@ -193,9 +193,25 @@ def creation_attributes
           attributes
         end
 
+        def instantiation_attributes(record)
+          attributes = {}
+
+          owners_name = owner.class.base_class.name.downcase
```

Comment:
That part of my solution is terrible and there probably is a better way to do it. I'll try to figure out how to find the association name in a better way. So far `inverse_reflection_for` does not seem like it does the job.